File: LeetCode Solutions/Problems/720. Longest Word in Dictionary.py
class Solution:
    def longestWord(self, words: List[str]) -> str:
        words.sort()
        wordset, former = set(['']), ''
        for word in words:
            if word[:-1] in wordset:
                former = word if len(word)>len(former) else former
                wordset.add(word)
        return former

File: LeetCode Solutions/Problems/720. Longest Word in Dictionary.cpp
class Solution {
public:
    string longestWord(vector<string>& words) {
        sort(words.begin(), words.end());
        unordered_set<string> ust;
        string former;
        for (auto &w : words){
            if (w.size() == 1 || ust.count(w.substr(0, w.size() - 1))){
                former = w.size() > former.size() ? w : former;  
                ust.insert(w);
          }
        }
        return former;
    }
};